server/services/ManagedIndexService.ts

```diff
@@ -83,59 +86,53 @@ export default class ManagedIndexService {
       };
 
       const managedIndexSorts: ManagedIndicesSort = { index: "managed_index.index", policyId: "managed_index.policy_id" };
-      const searchParams: RequestParams.Search = {
-        index: INDEX.OPENDISTRO_ISM_CONFIG,
-        seq_no_primary_term: true,
-        body: {
-          size,
-          from,
-          sort: managedIndexSorts[sortField] ? [{ [managedIndexSorts[sortField]]: sortDirection }] : [],
-          query: {
-            bool: {
-              filter: [{ exists: { field: "managed_index" } }],
-              must: getMustQuery("managed_index.name", search),
-            },
-          },
-        },
+      const explainParams = {
+        size,
+        from,
+        sortField: sortField ? managedIndexSorts[sortField] : null,
+        sortOrder: sortDirection,
+        queryString: search ? `*${search.split(" ").join("* *")}*` : null,
       };
 
       const { callAsCurrentUser: callWithRequest } = this.esDriver.asScoped(request);
-      const searchResponse: SearchResponse<any> = await callWithRequest("search", searchParams);
+      const explainAllResponse: ExplainAllResponse = await callWithRequest("ism.explainAll", explainParams);
 
-      const indices = searchResponse.hits.hits.map((hit) => hit._source.managed_index.index);
-      const totalManagedIndices = _.get(searchResponse, "hits.total.value", 0);
-
-      if (!indices.length) {
-        return response.custom({
-          statusCode: 200,
-          body: {
-            ok: true,
-            response: { managedIndices: [], totalManagedIndices: 0 },
-          },
+      const managedIndices: ManagedIndexItem[] = [];
+      for (const indexName in explainAllResponse) {
+        if (indexName == "total_managed_indices") continue;
+        const metadata = explainAllResponse[indexName] as ExplainAPIManagedIndexMetaData;
+        let policy, seqNo, primaryTerm, getResponse;
+        try {
+          getResponse = await callWithRequest("ism.getPolicy", { policyId: metadata.policy_id });
+        } catch (err) {
+          if (err.statusCode === 404 && err.body.error.reason === "Policy not found") {
+            console.log("managed index with not existing policy");
+          } else {
+            throw err;
+          }
+        }
+        policy = _.get(getResponse, "policy", null);
```

Comment:
Unclear what the behavior here is on the UI when the explain returns a response and the get fails and we have policy: null. I think before a policy could be null when it hadn't initialized yet, is that still possible? Can it be confusing now since policy: null on the UI can mean not initialized vs failed to get.

Reply:
Previously policy is from managed-index doc, now it's directly get from policy doc. So now, if policy is null,  will always show "Failed to load policy", it actually means policy not exist. If get policy call fail, it will throw an exception.

For initializing, check if metadata contain policy seqNo, if it's null, it means we are still initializing